netket/experimental/operator/_fermions_2nd.py

```diff
@@ -0,0 +1,621 @@
+from typing import List, Union, Optional
+import re
+from collections import defaultdict
+
+import numpy as np
+from numba import jit
+
+from netket.utils.types import DType
+from netket.hilbert import AbstractHilbert
+from netket.operator._discrete_operator import DiscreteOperator
+from netket.operator._pauli_strings import _count_of_locations
+from netket.hilbert.abstract_hilbert import AbstractHilbert
+
+from netket.experimental.hilbert import SpinOrbitalFermions
+import netket.experimental as nkx
+
+
+class FermionOperator2nd(DiscreteOperator):
+    def __init__(
+        self,
+        hilbert: AbstractHilbert,
+        terms: Union[List[str], List[List[List[int]]]],
+        weights: List[Union[float, complex]] = None,
+        constant: Union[float, complex] = 0.0,
+        dtype: DType = complex,
+    ):
+
+        r"""
+        Constructs a fermion operator given the single terms (set of creation/annihilation operators) in second quantization formalism.
+        This class can be initialized in the following form: ``FermionOperator2nd(hilbert, terms, weights ...)``.
+        The terms contain pairs of (idx, dagger), where the idx is the index in the output of the hilbert.all_states().
```

Comment:
I am slightly confused by `idx` here, do you mean this is not the orbital / lattice index but rather the index in the full hilbert space enumeration? why?

Reply:
This is the most general way of working. You just label the hilbert space enumeration. I want to avoid the difference between spin and orbitals at all cost here. I give the chance to work with that in netket.experimental.operator.fermion. It's only when you have a **hamiltonian** definition (for example) that the difference between spin and orbitals becomes important. What I can do is add more functionality to create a FermionOperator2nd with another staticmethod that does allow you to give spin projections.

Reply:
yes ok I think I still don't understand what `idx` is exactly, maybe it's just me though and the docstring is clear... not sure!

Reply:
more precisely, the thing that I don't understand is here 

> where the idx is the index in the output of the hilbert.all_states()

what hilbert is this? the many-fermion hilbert or the local hilbert?

Reply:
As far as I understand, `idx` here is and index in `hilbert.size` (which corresponds to an orbital). I think you should just rephrase that phrase to say 

```
The term contains Paris of `(idx, dagger)`, where `idx ∈ range(hilbert.size)` (it identifies an orbital) and `dagger` is a True/False flag determining if the operator is a creation or destruction operator.
```